railties/lib/rails/test_unit/test_requirer.rb

```diff
@@ -15,7 +15,7 @@ def require_files(patterns)
       private
         def expand_patterns(patterns)
           patterns.map do |arg|
-            arg = arg.gsub(/:(\d+)?$/, '')
+            arg = arg.gsub(/(:\d*)+?/, '')
```

Comment:
Fail to see how this matches at the end of the line? Won't we need `$` (or preferably `\z`) somewhere?